### DISCUSSION TITLE
Why Modelica doesn't provide a non ideal GearR2T and how to overcome that?

### DISCUSSION THREAD

**Discussion by s-celles** [OpenModelica/OpenModelica#11195] [Q&A]
Hello,

Sorry for double posting here https://stackoverflow.com/questions/77116771/why-modelica-doesnt-provide-a-non-ideal-gearr2t-and-how-to-overcome-that

I noticed that Modelica / OpenModelica Connection Editor provides a component called IdealGearR2T

[![idealGearR2T][1]][1]

as this name suggests it... it's an ideal component and so efficiciency in both direction is 1 (ie from rotational to translation and from translation to rotational).

I wonder why Modelica doesn't provide a non ideal GearR2T and how to overcome that ?

  [1]: https://i.stack.imgur.com/rW3sI.png

Kind regards

**Comment by s-celles** [comment]
Maybe a workaround could be to add a lossyGear before idealGearR2T but I don't know how to create my own block which contains lossyGear and idealGearR2T and could be named lossyGearR2T. Any idea?

**Reply by casella** [reply]
Please do experiment. If you get a component that tests positively, we can add it to MSL 4.2.0 in the near future.

**Reply by s-celles** [reply]
Thanks. But for experimenting that I need some help / guidance.

1. about inverting model https://github.com/OpenModelica/OpenModelica/discussions/11197 and see how this lossyGear + idealGearR2T behaves when swapping input and output
2. about how to build my "own block" which consists of a lossyGearR2T next to a lossyGear
